src/System.Xml.XmlSerializer/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -305,21 +303,27 @@ public override int GetHashCode()
 
     internal class TempAssemblyCache
     {
-        private Hashtable _cache = new Hashtable();
+        private Dictionary<TempAssemblyCacheKey, TempAssembly> _cache = new Dictionary<TempAssemblyCacheKey, TempAssembly>();
```

Comment:
Thanks for the insight. This means that this part could have issue with concurrent reading and writing to ```_cache```. Let me investigate this further to see if it needs to be fixed. Ref #7778